### ISSUE TITLE
incorrect ref escape check for expression body

### ISSUE THREAD

**Issue by bysdxt** [dotnet/roslyn#65388]
**Version Used**: 
.net7
sdk 7.0.100
**Steps to Reproduce**:

code:
```C#
namespace ConsoleApp1 {
    internal class Program {
        static void Main() {
        }
        static A f1(scoped ref int x) => new(ref x);
        static A f2(scoped ref int x) { return new(ref x); }
    }
    readonly ref struct A {
        public readonly ref int x;

        public A(ref int x) => this.x = ref x;
    }
}

```

**Diagnostic Id**:
CS9075
CS8347
**Expected Behavior**:
both f1 and f2 should be reported to CS9075 and CS8347
**Actual Behavior**:
just f2 is reported

![1](https://user-images.githubusercontent.com/13766766/201461725-f3ae6d20-7e31-4e14-a6c9-aaed318afbf4.png)

**Comment by jcouv** [comment]
@bysdxt Thanks for reporting this. It looks like it was fixed recently, as I now see CS9075 and CS8347 reported on both method bodies (see [sharplab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AJPtgFwKYBOMEANugAIBMFAjAOzoCwAUFugN4ttvk0Bs6AILoAZjQAUAZwDGAewAOeBOgJ4R2GDnQAPAJToAvAD50MPAHdxq9XoDcXbpl4DhIylLmLl1jVr0cKBjNLHzt0AF8HTEjWLFUIBFkYEgBPFTV0SRwCAFdpLWEojiKeAGZ0hKTU9PVcHXtmEqxycsErDLr/Y3QcAAsoSQA6bUMa+qLwoA===))

Tagging @cston to confirm (he may know precisely when this was fixed) and close.

**Comment by cston** [comment]
This appears to be https://github.com/dotnet/roslyn/issues/64045 which was fixed in https://github.com/dotnet/roslyn/pull/64976.
